openedx/core/djangoapps/courseware_api/utils.py

```diff
@@ -36,15 +35,14 @@ def get_celebrations_dict(user, enrollment, course, browser_timezone):
     if streak_length_to_celebrate:
         # We only want to offer the streak discount
         # if the course has not ended, is upgradeable and the user is not an enterprise learner
-        discount_enabled = COURSEWARE_MFE_MILESTONES_STREAK_DISCOUNT.is_enabled()
```

Comment:
Pushed this check to can_show_streak_discount_coupon so all the logic for whether the coupon is shown is in the same place.